CRM/Upgrade/Incremental/php/FiveThirtySeven.php

```diff
@@ -139,4 +140,45 @@ public static function updateDBDefaultsForContributionRecur(CRM_Queue_TaskContex
     return TRUE;
   }
 
+  /**
+   * Install recaptcha extension.
+   *
+   * This feature is restructured as a core extension - which will eventually allow us to replace/remove the
+   * reCAPTCHA implementation
+   * @param \CRM_Queue_TaskContext $ctx
+   *
+   * @return bool
+   * @throws \CRM_Core_Exception
+   */
+  public static function installReCaptchaExtension(CRM_Queue_TaskContext $ctx) {
+    // Install via direct SQL manipulation. Note that:
+    // (1) This extension has no activation logic.
+    // (2) On new installs, the extension is activated purely via default SQL INSERT.
+    // (3) Caches are flushed at the end of the upgrade.
+    // ($) Over long term, upgrade steps are more reliable in SQL. API/BAO sometimes don't work mid-upgrade.
+    $insert = CRM_Utils_SQL_Insert::into('civicrm_extension')->row([
+      'type' => 'module',
+      'full_name' => 'recaptcha',
+      'name' => 'reCAPTCHA',
+      'label' => 'reCAPTCHA',
+      'file' => 'recaptcha',
+      'schema_version' => NULL,
+      'is_active' => 1,
+    ]);
+    CRM_Core_DAO::executeQuery($insert->usingReplace()->toSQL());
+
+    CRM_Core_DAO::executeQuery('
+UPDATE civicrm_navigation
+SET name="Misc (Undelete, PDFs, Limits, Logging, etc.)"
+WHERE name="Misc (Undelete, PDFs, Limits, Logging, Captcha, etc.)"
+');
+    CRM_Core_DAO::executeQuery('
+UPDATE civicrm_navigation
+SET label="Misc (Undelete, PDFs, Limits, Logging, etc.)"
+WHERE label="Misc (Undelete, PDFs, Limits, Logging, Captcha, etc.)"
```

Comment:
This isn't right. Labels should be translated, and this label is English-only. The WHERE should be keyed to the name, and the new label should be in `ts()`.

Reply:
When I first looked at this, I had the same thought and started to draft a comment with a suggestion along those lines. Then I decided to test instead of speculate. And... the translation works differently for labels on nav-items. This is an pre-existing condition. Do this:

* Install with a French dataset
* Look in the nav-menu -- you'll see the label in French
* Look in the DB -- you'll see the label in English

It seems to have the effect of displaying menu items with `ts($navItem.label)`. This strikes an odd but functional balance:

* In the typical case of a default label, the menus will be automatically converted to the user's language. Different users may display UI in different languages.
* In the atypical case of customized labels, the menu will show the customization. The customization is unlikely to be translated.

This works out for most configurations which are (a) pure-default labels on single-language, (b)  pure-default labels on multi-language, or (c) custom defaults on single-language. It breaks down a little with custom-labels on multi-language... but TBH it's hard for me to feel too strongly that the behavior is *wrong*

Of course, I agree that this feels a little insane. It's the pre-existing condition.